app/src/main/java/com/readrops/app/item/view/ItemWebView.kt

```diff
@@ -78,17 +80,22 @@ class ItemWebView(
     }
 
     private fun formatText(itemWithFeed: ItemWithFeed): String {
-        return if (itemWithFeed.item.text != null) {
-            val document = if (itemWithFeed.websiteUrl != null) Jsoup.parse(
-                Parser.unescapeEntities(itemWithFeed.item.text, false), itemWithFeed.websiteUrl
-            ) else Jsoup.parse(
-                Parser.unescapeEntities(itemWithFeed.item.text, false)
-            )
-
-            document.select("div,span").forEach { it.clearAttributes() }
-            return document.body().html()
+        val text = itemWithFeed.item.text ?: return ""
+        val unescapedText = Parser.unescapeEntities(text, false)
+        val document = if (itemWithFeed.websiteUrl != null) {
+            Jsoup.parse(unescapedText, itemWithFeed.websiteUrl!!)
+        } else {
+            Jsoup.parse(unescapedText)
+        }
+        // If body has no tags or all tags are unknown (and therefore likely not HTML tags at all),
+        // treat the whole thing as plain text and convert it to HTML turning newlines into <br>/<p> tags
+        val body = document.body()
+        val isPlainText = body.stream().skip(1).allMatch { !it.tag().isKnownTag }
+        return if (isPlainText) {
+            HtmlCompat.toHtml(SpannedString(unescapedText), HtmlCompat.TO_HTML_PARAGRAPH_LINES_CONSECUTIVE)
```

Comment:
According to `HtmlCompat.toHtml` documentation, `options` parameter is ignored below API 24. As Readrops supports API down to 21, would we have an alternative here?

Reply:
Old method that doesn't take options has the same behaviour as the new one with TO_HTML_PARAGRAPH_LINES_CONSECUTIVE, so it will do what we want to (I just checked on API 21 emulator).

Reply:
Perfect, thanks for this!